server/container_update_resources.go

```diff
@@ -27,13 +27,21 @@ func (s *Server) UpdateContainerResources(ctx context.Context, req *types.Update
 	}
 
 	if req.Linux != nil {
-		resources := toOCIResources(req.Linux)
+		updated, err := s.nri.updateContainer(ctx, c, req.Linux)
+		if err != nil {
+			return nil, err
+		}
+		resources := toOCIResources(updated)
 		if err := s.Runtime().UpdateContainer(ctx, c, resources); err != nil {
 			return nil, err
 		}
 
 		// update memory store with updated resources
 		s.UpdateContainerLinuxResources(c, resources)
+
+		if err := s.nri.postUpdateContainer(ctx, c); err != nil {
+			log.Errorf(ctx, "NRI container post-update failed: %v", err)
```

Comment:
Again, there's an inconsistency here -- earlier it's mostly either `log.Warnf` or returning an error, in here you logging (but not returning) an error.

Reply:
It's a post-update NRI call which can't alter state (inside the runtime). Therefore it does not cause the update to return with an error.

Now to be honest, I'm not 100% sure that the current approach (treat only the errors in potentially state-altering calls as an error of the original CRI call)  is the right one, or if it is always the right one. IMO, we haven't had/seen enough use cases where an NRI plugin would carry out extra-OCI actions (as opposed or in addition to altering container configuration via the NRI API) to be able to really say. We'll need to figure this out as we gain more experience ourselves and feedback from others.